main.js

```diff
@@ -64,41 +66,82 @@ function createLabelRenderer() {
     return labelRenderer;
 }
 
+function createPlayerInfoLabel() {
+    const div = document.createElement('div');
+    div.id = 'player-info';
+    div.className = 'label';
+    div.textContent = `Player turn: ${currentPlayer}`;
+    currentPlayerLabel = new CSS2DObject(div);
+    currentPlayerLabel.position.set(0, 1, 0);
+    scene.add(currentPlayerLabel);
+}
+
 function initializeEventListeners(camera, renderer, labelRenderer, gameBoard, raycaster, plane, updateSizes) {
 
+    let isDragging = false;
+
+    const handleMouseDown = function(event) {
+        isDragging = false;
+    };
+
+    const handleMouseMove = function(event) {
+        isDragging = true;
+    };
+
+    const handleMouseUp = function(event) {
+        if (!isDragging) {
+            handlePlayerTurn(event);
+        }
+        isDragging = false;
+    };
+
     const handlePlayerTurn = function (event) {
-        try {
-            const mouse = new THREE.Vector2();
-            mouse.x = (event.clientX / window.innerWidth) * 2 - 1;
-            mouse.y = -(event.clientY / window.innerHeight) * 2 + 1;
-            raycaster.setFromCamera(mouse, camera);
-            const intersects = raycaster.intersectObjects([plane]);
-            if (intersects.length > 0) {
-                let intersectPoint = intersects[0].point;
-                let gridX = Math.round(intersectPoint.x);
-                let gridY = Math.round(Math.abs(intersectPoint.y));
-                // check if grid indexes are within the board size
-                if (gridX >= 0 && gridX <= BOARD_SIZE && gridY >= 0 && gridY <= BOARD_SIZE) {
-                    // Draw a new stone only if the grid cell is currently empty
-                    if (gameBoard[gridY][gridX] === null) {
-                        drawStone(gridX, gridY, currentPlayer, gameBoard);
-                        currentPlayer = currentPlayer === 'black' ? 'white' : 'black';
-                        document.getElementById('info').innerText = `Player turn: ${currentPlayer}`;
+            try {
+                const mouse = new THREE.Vector2();
+                mouse.x = (event.clientX / window.innerWidth) * 2 - 1;
+                mouse.y = -(event.clientY / window.innerHeight) * 2 + 1;
+                raycaster.setFromCamera(mouse, camera);
+                const intersects = raycaster.intersectObjects([plane]);
+
+                if (intersects.length === 0) {
+                    return;  // If not clicking on game board, do nothing.
+                }
+
+                if (intersects.length > 0) {
+                    let intersectPoint = intersects[0].point;
+                    let gridX = Math.round(intersectPoint.x);
+                    let gridY = Math.round(Math.abs(intersectPoint.y));
+                    // console.log("gridX: " + gridX + " gridY: " + gridY);
```

Comment:
Best to delete old console logs (and probably old code) if you're going to be sending them for review. Otherwise they are just extra noise. I don't mind for our situation, but just be aware that in a professional setting, this is frowned upon.